catalog_publish.java

```diff
@@ -261,35 +262,47 @@ boolean processExtension(Path extensionYaml) {
             throws IOException {
-        URI platformJson;
-        if (classifier == null) {
-            platformJson = URI.create(MessageFormat.format("{0}{1}/{2}/{3}/{2}-{3}.json",
-                    Objects.toString(repository, MAVEN_CENTRAL),
-                    groupId.replace('.', '/'),
-                    artifactId,
-                    version));
-        } else {
-            // https://repo1.maven.org/maven2/io/quarkus/quarkus-bom-quarkus-platform-descriptor/1.13.0.Final/quarkus-bom-quarkus-platform-descriptor-1.13.0.Final-1.13.0.Final.json
-            platformJson = URI.create(MessageFormat.format("{0}{1}/{2}/{3}/{2}-{4}-{3}.json",
-                    Objects.toString(repository, MAVEN_CENTRAL),
-                    groupId.replace('.', '/'),
-                    artifactId,
-                    version,
-                    classifier));
-        }
-        if ("file".equals(platformJson.getScheme())) {
-            var path = Path.of(platformJson);
-            if (!Files.exists(path)) {
-                throw new IOException(path + " does not exist");
+        List<URI> triedUris = new ArrayList<>();
+        for(String repository: repositories) {
+            URI platformJson;
+            if (classifier == null) {
+                platformJson = URI.create(MessageFormat.format("{0}{1}/{2}/{3}/{2}-{3}.json",
+                        Objects.toString(repository, MAVEN_CENTRAL),
+                        groupId.replace('.', '/'),
+                        artifactId,
+                        version));
+            } else {
+                // https://repo1.maven.org/maven2/io/quarkus/quarkus-bom-quarkus-platform-descriptor/1.13.0.Final/quarkus-bom-quarkus-platform-descriptor-1.13.0.Final-1.13.0.Final.json
+                platformJson = URI.create(MessageFormat.format("{0}{1}/{2}/{3}/{2}-{4}-{3}.json",
+                        Objects.toString(repository, MAVEN_CENTRAL),
+                        groupId.replace('.', '/'),
+                        artifactId,
+                        version,
+                        classifier));
+            }
+            triedUris.add(platformJson);
+            if ("file".equals(platformJson.getScheme())) {
+                var path = Path.of(platformJson);
+                if (!Files.exists(path)) {
+                    throw new IOException(path + " does not exist");
+                }
+                return Files.readAllBytes(path);
+            }
+            try (CloseableHttpClient httpClient = createHttpClient()) {
+                CloseableHttpResponse response = httpClient.execute(new HttpGet(platformJson));
+                try (InputStream is = response.getEntity().getContent()) {
+                    if (response.getStatusLine().getStatusCode() != 200) {
+                        log.info("Can't get the extension catalog from " + platformJson + ", server responded: " + new String(is.readAllBytes()));
```

Comment:
```suggestion
                        log.warn("Can't get the extension catalog from " + platformJson + ", server responded: " + new String(is.readAllBytes()));
```

Reply:
I don't think this is necessarily  a warning, when multiple repos are declared, it is expected that not all of them will have the necessary artifact. Only if we don't find it in any of them, then we throw an error, and that error contains all URIs that we tried.

Reply:
Hmm, so it's more of a `debug` then?

Reply:
Then it wouldn't be shown by default, and I think we don't log all that much stuff that we need to filter what we log :P